### PR TITLE
Improve docs of softmax_cross_entropy

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
This is PR for improving docs of functions/links. Related issue: #2182

Modified functions/links:

- functions/loss/softmax_cross_entropy.py

### CODE REVIEW

**Comment by mitmul** [comment]
Jenkins, test this please

**Comment by mitmul** [comment]
@keisuke-umezawa Jenkins tests failed. Could you rebase it to the current master?

**Comment by keisuke-umezawa** [comment]
@mitmul I rebased this branch.

File: chainer/functions/loss/softmax_cross_entropy.py [hunk 1]
```diff
@@ -223,17 +223,21 @@ def backward_gpu(self, inputs, grad_outputs):
 def softmax_cross_entropy(
         x, t, normalize=True, cache_score=True, class_weight=None,
         ignore_label=-1, reduce='mean'):
-    """Computes cross entropy loss for pre-softmax activations.
+    """Computes cross entropy loss after softmax activations.
```
Comment (rezoo) [inline chainer/functions/loss/softmax_cross_entropy.py]: It seems that the previous sentence is better; in your sentence, it is a bit confusing that we should apply "softmax" functions before calling `softmax_cross_entropy`.

File: chainer/functions/loss/softmax_cross_entropy.py [hunk 2]
```diff
@@ -261,14 +266,32 @@ class weight ``class_weight[i]`` is actually multiplied to
...
+        >>> t = np.array([3, 0]).astype('i')
+        >>> t
+        array([3, 0], dtype=int32)
+        >>> y = F.softmax_cross_entropy(x, t)
+        >>> y
+        variable(0.4401897192001343)
+        >>> log_softmax = -F.log_softmax(x)
+        >>> expected_loss = np.mean([log_softmax[row, column].data \
+for row, column in enumerate(t)])
+        >>> y.data == expected_loss
```
Comment (rezoo) [inline chainer/functions/loss/softmax_cross_entropy.py]: How about using `y.array` instead of `y.data`?

File: chainer/functions/loss/softmax_cross_entropy.py (hunk 1 above)
Comment (keisuke-umezawa) [reply chainer/functions/loss/softmax_cross_entropy.py]: revereted it.

File: chainer/functions/loss/softmax_cross_entropy.py (hunk 2 above)
Comment (keisuke-umezawa) [reply chainer/functions/loss/softmax_cross_entropy.py]: ok. I used `y.array`

**Comment by keisuke-umezawa** [comment]
@rezoo I fixed them!

**Comment by rezoo** [comment]
Thank you. LGTM.
